map.go

```diff
@@ -0,0 +1,135 @@
+package validation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+var (
+	// ErrNotMap is the error that the value being validated is not a map.
+	ErrNotMap = errors.New("only a map can be validated")
+
+	// ErrKeyMissing is the error returned in case of an incorrect key type.
+	ErrKeyWrongType = NewError("validation_key_wrong_type", "key not the correct type")
+
+	// ErrKeyNotFound is the error returned in case of a missing key.
+	ErrKeyNotFound = NewError("validation_key_missing", "key not found")
+
+	// ErrKeyNotExpected is the error returned in case of an unexpected key.
```

Comment:
There are some mismatch between the var name in the comment and the actual var name.